### DISCUSSION TITLE
Side TOC and search capabilities

### DISCUSSION THREAD

**Discussion by felwind14** [dita-ot/dita-ot#4516] [Q&A]
HI everybody,

I am quite new to DITA-OT side of things so naturally I am struggling a bit with this issues:

- Left TOC not displayed in HTML based outputs
- Search not working in the same HTML output

This is a bit of the background of my case:

I have my project structured in oxygen and I organised my project like this:

- project.xpr
- manual_folder:
  - topicsfolder
  - mastermap.ditamap
  - transformation.xml

When I run the transformation xml with this parameters:

     <publication transtype="html5-bootstrap">
         <param name="args.draft" value="no"/>
         <param name="args.figurelink.style" value="NUMBER"/>
         <param name="args.tablelink.style" value="NUMBER"/>
         <param name="args.output.base" value="2D-Commands_Objects"/>
         <param name="args.copycss" value="yes"/>
         <param name="args.csspath" value="css"/>
         <param name="args.cssroot" path="../resources"/>
         <param name="html5.toc.generate" value="no"/>
         <param name="nav-toc" value="list-group-partial"/>
         <param name="args.breadcrumbs" value="yes"/>

there is an output html without the side TOC and the search (also  I tried switching to yes   <param name="html5.toc.generate" value="yes"/> and also didn't work. So I got told this: "This is mostly an issue with the structure of the DITA project. Unfortunately, the master map has to be at the root of the project in order for the search, table of contents, and other webhelp features to work "

So I tried to put in an upper level:

- project.xpr
- manual_folder:
  - topicsfolder
  - transformation.xml
-mastermap.ditamap 

and the side TOC shows up, but the search continues without working.

I really think it should allow me the first project structure I have seen in other youtube videos. Could you help me how to approach the issue to get the desired result?

**Comment by raducoravu** [comment]
I see you are using a DITA OT project file to run the transformation.
Assuming you are using the DITA OT publishing engine bundled with Oxygen, Oxygen has its own "webhelp-responsive" transformation type which generates HTML output with a TOC and search capabilities. But if you have to use the "html5-bootstrap" transformation type, I'm not too familiar with it so I cannot help here.

**Reply by felwind14** [reply]
@raducoravu In your perception seems logic that the .ditamap needs to be in the same level than the .xpr to work as expected?

**Reply by raducoravu** [reply]
The project structure looks good to me, the DITA Map is in a folder above the DITA topics. Where the project.xpr is does not matter, the project.xpr is an Oxygen specific project configuration file, the DITA OT does not use it for anything.

**Comment by lakokkonen** [comment]
FYI, the html5-bootstrap's (also known as [DITA Bootstrap](https://infotexture.github.io/dita-bootstrap/)) search engine doesn't work with local files. You need to use a local HTTP server for that.  At least that's my experience, and assuming you're using the [Lunr Search](https://github.com/infotexture/dita-bootstrap.lunr) described in the docs.

Could this be the reason for the search not working?
